packages/ui/src/badge/types.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { type ComponentProps } from '../utils/types';
+
+export interface BadgeProps extends ComponentProps< 'span' > {
+	/**
+	 * The text to display in the badge.
+	 */
+	children: string;
```

Comment:
I think we discussed how to approach this elsewhere but I don't recall the outcome 🙈 Please feel free to push changes.